src/opentimelineio/clip.h

```diff
@@ -9,24 +9,39 @@ namespace opentimelineio { namespace OPENTIMELINEIO_VERSION {
 class Clip : public Item
 {
 public:
+    static char constexpr default_media_key[] = "DEFAULT_MEDIA";
+
     struct Schema
     {
         static auto constexpr name   = "Clip";
-        static int constexpr version = 1;
+        static int constexpr version = 2;
     };
 
     using Parent = Item;
 
     Clip(
-        std::string const&         name            = std::string(),
-        MediaReference*            media_reference = nullptr,
-        optional<TimeRange> const& source_range    = nullopt,
-        AnyDictionary const&       metadata        = AnyDictionary());
-
-    void set_media_reference(MediaReference* media_reference);
+        std::string const&         name               = std::string(),
+        MediaReference*            media_reference    = nullptr,
+        optional<TimeRange> const& source_range       = nullopt,
+        AnyDictionary const&       metadata           = AnyDictionary(),
+        std::string const& active_media_reference_key = default_media_key);
 
+    void            set_media_reference(MediaReference* media_reference);
     MediaReference* media_reference() const noexcept;
 
+    using MediaReferences = std::map<std::string, MediaReference*>;
+
+    MediaReferences media_references() const noexcept;
+    void            set_media_references(
+                   MediaReferences const& media_references,
+                   std::string const&     new_active_key = std::string(),
```

Comment:
The one remaining question I have after the TSC meeting.  I asked it already on slack, so I'll repost here:

 I have a brief followup question from yesterday's TSC meeting.  When we discussed modifying the media_refererence and media_reference_key setters so that they will enforce that the active key is never invalidated, something that occurred to me after the meeting is that this would make it very difficult to remove the active key.  For example, if my keys were ['a', 'b'] with 'a' being active and I wanted to change for ['c', 'd'] with 'c' being active, I would need to do leave the active key in there temporarily, then change the active key, then reset the values with the old active key removed. Something like this:
clip.set_media_references( {{"a": dummy_ref},{"b", ref1}, {"c", ref2}} );
clip.set_active_media_reference_key("b");
clip.set_media_references( {{"b", ref1}, {"c", ref2}} );
Needless to say that is quite tedious.  I think the best solution for this is to allow the user to pass a new active_key to the set_media_reference call.  Something like this:
clip.set_media_references( {{"b", ref1}, {"c", ref2}}, "b");
I am wondering the second parameter should be optional or not and if unset we leave the key as is?
If we make it mandatory it would be trivial but maybe a bit tedious to leave it as is:
clip.set_media_refererences( {{"a", ref1}, {"b", ref2}}, clip.get_active_media_reference_key() ); 
And if we do leave it optional is the behaviour of defaulting to an empty string make the intent clear enough?
void set_media_references(MediaReferences const& media_references,
                          std::string const& new_active_key = std::string(),
                          ErrorStatus* error_status = nullptr);
This would of course assign special meaning to empty string as well, preventing the user from using empty string as a key in the dictionary that they're setting.  Obviously, an edge-case but something to consider. I think using std::optional<std::string> is a more clear way of doing it but that's not done anywhere in the API currrently so there's no mechanism for serializing that at the moment (I could add it if desired).  Alternatively, it could be done the old-school way with a pointer to std::string defaulting to nullptr.

Reply:
For a first draft, I lean to simple-as-possible. Why don't we try making it non-optional and discover whether it's actually too clumsy in practice? This looks perfectly fine to me: `clip.set_media_references( {{"b", ref1}, {"c", ref2}}, "b");`

Reply:
I am presuming that the simplest case is actually the common one...